src/org/opensolaris/opengrok/history/GitRepository.java

```diff
@@ -544,64 +505,32 @@ private TagEntry buildTagEntry(File directory, String tags) throws HistoryExcept
                 + "Date:%at");
         argv.add("-r");
         argv.add(tags + "^.." + tags);
-        ProcessBuilder pb = new ProcessBuilder(argv);
-        pb.directory(directory);
-        Process process;
-        process = pb.start();
-
-        try (BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
-            String line;
-            while ((line = in.readLine()) != null) {
-                if (line.startsWith("commit")) {
-                    String parts[] = line.split(":");
-                    if (parts.length < 2) {
-                        throw new HistoryException("Tag line contains more than 2 columns: " + line);
-                    }
-                    hash = parts[1];
-                }
-                if (line.startsWith("Date")) {
-                    String parts[] = line.split(":");
-                    if (parts.length < 2) {
-                        throw new HistoryException("Tag line contains more than 2 columns: " + line);
-                    }
-                    date = new Date((long) (Integer.parseInt(parts[1])) * 1000);
-                }
-            }
-        }
-
-        try {
-            process.exitValue();
-        } catch (IllegalThreadStateException e) {
-            // the process is still running??? just kill it..
-            process.destroy();
-        }
-
-        // Git can have tags not pointing to any commit, but tree instead
-        // Lets use Unix timestamp of 0 for such commits
-        if (date == null) {
-            date = new Date(0);
-        }
-        TagEntry result = new GitTagEntry(hash, date, tags);
-        return result;
+        
+        Executor executor = new Executor(argv, directory,
+                RuntimeEnvironment.getInstance().getInteractiveCommandTimeout());
+        GitTagParser parser = new GitTagParser(tags);
+        executor.exec(true, parser);
+        return parser.getEntries().first();
     }
 
     @Override
-    protected void buildTagList(File directory) {
+    protected void buildTagList(File directory, boolean interactive) {
```

Comment:
Wondering whether the new `interactive` parameter should possibly be used in this method.

Reply:
good catch, yes. Also, I've just hit this when testing full reindex on a busy machine where I deliberately set the interactive timeout to 3 seconds.